scrapy/pipelines/files.py

```diff
@@ -269,24 +269,19 @@ class FilesPipeline(MediaPipeline):
 
     MEDIA_NAME = "file"
     EXPIRES = 90
-    STORE_SCHEMES = {
-        '': FSFilesStore,
-        'file': FSFilesStore,
-        's3': S3FilesStore,
-        'gs': GCSFilesStore,
-    }
     DEFAULT_FILES_URLS_FIELD = 'file_urls'
     DEFAULT_FILES_RESULT_FIELD = 'files'
+    STORAGES = {}
 
-    def __init__(self, store_uri, download_func=None, settings=None):
-        if not store_uri:
+    def __init__(self, urifmt, download_func=None, settings=None):
+        if not urifmt:
             raise NotConfigured
 
         if isinstance(settings, dict) or settings is None:
             settings = Settings(settings)
 
         cls_name = "FilesPipeline"
-        self.store = self._get_store(store_uri)
```

Comment:
This is almost backwards compatible, because self.store is set in open_spider method. 
Do you need to use open_spider because crawler.spider is not set when FilesPipeline.from_crawler is called? (haven't checked it)